lib/main.py

```diff
@@ -31,11 +31,11 @@ def waitForThreads():
                     util.DEBUG_LOG('Main: Waiting on: {0}...'.format(t.name))
                     if isinstance(t, threading._Timer):
                         t.cancel()
+
+                    try:
                         t.join()
```

Comment:
I think this should be `t.join(timeout=10.0)`. I know that the threads should exit completely before exiting the app, but I think the user comes first here. `10.0` is the default connection timeout set by P4K.

Reply:
The main issue with not waiting for threads to exit is that if a thread is running on Kodi shutdown, Kodi will hang or crash. Waiting for them all here makes sure if some thread is hanging (like with the issue you're fixing here) that it's obvious that there's an issue with the add-on rather than a mysterious Kodi hang. On the flip side if we're confident that any thread still alive after 10 seconds here is going to exit soon then adding the timeout could make sense.

Reply:
Hmm. The logic is sound, but then we should have ```
try:
    t.join()
except:
    util.ERROR()
```?